### PR TITLE
Updated the package source resource to accept credentials and updated…

### PR DESCRIPTION
**Pull request by johnmccrae**
… the code to be more readable

Signed-off-by: John McCrae <john.mccrae@progress.com>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/resource/powershell_package_source.rb [hunk 1]
```diff
@@ -117,10 +117,10 @@ class PowershellPackageSource < Chef::Resource
         description: "A label that names your package source.",
         name_property: true
 
-      property :new_name, String,
-        description: "Used when updating the name of a NON-PSRepository"
+      property :new_name, introduced: "17.5.23", String,
+        description: "Used to change the name of a standard PackageSource."
 
-      property :source_location, String,
+      property :source_location, introduced: "17.5.23", String,
```
Comment (tas50) [inline lib/chef/resource/powershell_package_source.rb:123]:
```suggestion
      property :source_location, introduced: "17.6", String,
```

File: lib/chef/resource/powershell_package_source.rb [hunk 2]
```diff
@@ -117,10 +117,10 @@ class PowershellPackageSource < Chef::Resource
         description: "A label that names your package source.",
         name_property: true
 
-      property :new_name, String,
-        description: "Used when updating the name of a NON-PSRepository"
+      property :new_name, introduced: "17.5.23", String,
```
Comment (tas50) [inline lib/chef/resource/powershell_package_source.rb:120]: If these properties already existed then they don't need introduced values

File: lib/chef/resource/powershell_package_source.rb (hunk 1 above)
Comment (tas50) [inline lib/chef/resource/powershell_package_source.rb:123]: If these properties already existed then they don't need introduced values

File: lib/chef/resource/powershell_package_source.rb (hunk 2 above)
Comment (tas50) [inline lib/chef/resource/powershell_package_source.rb:120]:
```suggestion
      property :new_name, introduced: "17.6", String,
```

File: lib/chef/resource/powershell_package_source.rb (hunk 1 above)
Comment (tas50) [inline lib/chef/resource/powershell_package_source.rb:123]:
```suggestion
      property :source_location, introduced: "17.6", String,
```

File: lib/chef/resource/powershell_package_source.rb [hunk 3]
```diff
@@ -138,10 +138,10 @@ class PowershellPackageSource < Chef::Resource
         description: "Whether or not to trust packages from this source. Used when creating a NON-PSRepository Package Source",
         default: false
 
-      property :user_name, String,
+      property :user, introduced: "17.5.23", String,
```
Comment (tas50) [inline lib/chef/resource/powershell_package_source.rb:141]:
```suggestion
      property :user, introduced: "17.6", String,
```

File: lib/chef/resource/powershell_package_source.rb [hunk 4]
```diff
@@ -138,10 +138,10 @@ class PowershellPackageSource < Chef::Resource
...
         description: "A username that, as part of a credential object, is used to register a repository or other package source with."
 
-      property :user_pass, String,
+      property :password, introduced: "17.5.23", String,
```
Comment (tas50) [inline lib/chef/resource/powershell_package_source.rb:144]:
```suggestion
      property :password, introduced: "17.6", String,
```

File: lib/chef/resource/powershell_package_source.rb [hunk 5]
```diff
@@ -117,10 +117,10 @@ class PowershellPackageSource < Chef::Resource
         description: "A label that names your package source.",
         name_property: true
 
-      property :new_name, String,
-        description: "Used when updating the name of a NON-PSRepository"
+      property :new_name, introduced: "17.6", String,
+        description: "Used to change the name of a standard PackageSource."
```
Comment (IanMadd) [inline lib/chef/resource/powershell_package_source.rb:121]:
```suggestion
        description: "Used to change the name of a standard package source."
```

**Comment by IanMadd** [comment]
Can we change the description on line 28 to: `Use the **powershell_package_source** resource to register a PowerShell package source and a Powershell package provider.`

**Pull request merged**